### PR TITLE
PHP8 Support

### PR DESCRIPTION
**Pull request by KartaviK**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ php:
     - 7.2
     - 7.3
     - 7.4
+    - 8.0
```
Comment (derrabus) [inline .travis.yml:9]:
```suggestion
    - '8.0snapshot'
```

Travis does not have stable 8.0 builds yet.

**Comment by VeryStrongFingers** [comment]
@derrabus looks like it's in there now https://travis-ci.community/t/php-8-0-missing/10132/24

**Comment by derrabus** [comment]
@VeryStrongFingers Yes, it has been added in the meantime.

**Comment by tarlepp** [comment]
Any update about this?

**Comment by Slamdunk** [comment]
From the failing CI, I guess we need to wait for https://github.com/doctrine/mongodb-odm/commit/758bc3bf45d3b3273c85da3819cab29f5f432a61 to be released first.

**Comment by svycka** [comment]
> From the failing CI, I guess we need to wait for [doctrine/mongodb-odm@758bc3b](https://github.com/doctrine/mongodb-odm/commit/758bc3bf45d3b3273c85da3819cab29f5f432a61) to be released first.

or we could run `composer install --ignore-platform-reqs`

**Comment by Slamdunk** [comment]
The end user should not do that, and neither should we for dependencies

**Comment by svycka** [comment]
> The end user should not do that, and neither should we for dependencies

maybe when testing this is ok and also we could release and dont wait for doctrine-orm and whn its ready users wont have to wait for our release. 

at least Laminas framework does this for many its own libraries maybe its not taht bad? https://github.com/orgs/laminas/projects/11

**Comment by kluevandrew** [comment]
Any updates on this?

File: .travis.yml [hunk 2]
```diff
@@ -6,6 +6,7 @@ php:
     - 7.2
     - 7.3
     - 7.4
+    - '8.0snapshot'
```
Comment (derrabus) [inline .travis.yml:9]:
```suggestion
    - 8.0
```

We can change this back, now that Travis provides stable 8.0 releases.

**Comment by AkenRoberts** [comment]
Thank you, and apologies for the delay. PHP 8 compatibility is now in 3.x (3.0.3 specifically). Should be in Packagist shortly. ❤️

**Comment by mmarton** [comment]
@AkenRoberts how much is shortly? seems like something went wrong on packagist.

**Comment by AkenRoberts** [comment]
I'm not sure. I don't own this repo and auto-publish isn't set up, and I guess Packagist only lets me manually refresh it once a day (and I used that when I goofed up the 3.0.1 release that this was supposed to be 🤦‍♂).

**Comment by derrabus** [comment]
3.0.3 is on Packagist now.

**Pull request closed without merging**